lib/dispatcher/pool.js

```diff
@@ -71,6 +72,21 @@ class Pool extends PoolBase {
       : undefined
     this[kFactory] = factory
 
+    this.on('connect', (origin, targets) => {
+      if (clientLifetime != null && clientLifetime > 0) {
+        setTimeout(() => {
```

Comment:
I am a little concerned that creating N timers isn't the most efficient way of handling this.  Another option would be to mark the client with a create time and then validate it before operations to make sure it hasn't expired.  If it has, we perform the logic here.

Reply:
I'd prefer to explore that option; under high level of clients, this can create unnecessary overhead